tests/reload/remove-add-alter-task/suite.rc

```diff
@@ -14,18 +14,26 @@
       script = false
    [[reloader]]
       script = """
-sed -i "12s/\(graph = reloader\).*/\1/" $CYLC_SUITE_DEF_PATH/suite.rc
-cylc reload $CYLC_SUITE_NAME
-sleep 5
-sed -i "12s/\(graph = reloader\)/\1 => remove_add_alter_me/" $CYLC_SUITE_DEF_PATH/suite.rc
-cylc reload $CYLC_SUITE_NAME
-sleep 5
-cylc insert $CYLC_SUITE_NAME remove_add_alter_me.1
-sleep 5
-cat >>$CYLC_SUITE_DEF_PATH/suite.rc <<'__RUNTIME__'
+do_reload() {
+    cylc reload "${CYLC_SUITE_NAME}"
+    while test "$(grep -cF 'Reload completed' "${CYLC_SUITE_LOG_DIR}/log")" -ne "$1"
+    do
+        sleep 1
+    done
```

Comment:
Might want to add some form of timeout to these tests prevent them from hanging.

Reply:
Use the `poll` function from `test_header` - it now aborts after a minute if condition not met.

Reply:
Yes, if there is an easy way to expose the test battery poll function in the job scripts.

Reply:
Set `PATH` at the suite level?

Reply:
No straightforward. It is a shell function in the `test_header`.

Reply:
You can always resort to brute force (like I did recently): https://github.com/cylc/cylc-flow/blob/master/tests/reload/22-remove-task-cycling.t#L47-L63

Reply:
For the purpose of this change, we should have a way to share the shell function, because the pattern is repeated in many places. I wonder if we should just add the function to `cylc/flow/etc/job.sh`? Maybe users can make use of a poll function in their task jobs as well?

Reply:
There are multiple test functions it might be useful to run from within a suite. How about moving the functions out of the test header into another file which can easily be sourced in a `pre-script` or something like that.

Reply:
Yes, a library of shell functions is definitely useful, but probably outside the scope of this PR.

Reply:
New issue for this: #3346

Reply:
I have added a function to `cylc/flow/etc/job.sh` to allow job scripts that have the same *poll suite log* pattern to simple call a function. We can generalise this as part of #3346.